dashboard/app/models/census/census_summary.rb

```diff
@@ -26,6 +26,72 @@ class Census::CensusSummary < ApplicationRecord
     HISTORICAL_YES: "HY",
     HISTORICAL_NO: "HN",
     HISTORICAL_MAYBE: "HM",
+    EXCLUDED: "E",
+    UNKNOWN: "U",
   }.freeze
   enum teaches_cs: TEACHES
+
+  # Use the zero byte as the quote character to allow importing double quotes
+  #   via http://stackoverflow.com/questions/8073920/importing-csv-quoting-error-is-driving-me-nuts
+  CSV_IMPORT_OPTIONS = {col_sep: ",", headers: true, encoding: 'bom|utf-8'}.freeze
+
+  # Seeds all the data from the S3 file.
+  # @param options [Hash] Optional map of options.
+  def self.seed_all(options = {})
+    options[:stub_school_data] = CDO.stub_school_data unless options.key?(:stub_school_data)
+
+    unless options[:stub_school_data]
+      Census::CensusSummary.transaction do
+        CDO.log.info "Seeding census summary data."
+        AWS::S3.seed_from_file('cdo-census', "access-report-data-files-2023/final_csv/final_csv_2023_09_19.csv") do |filename|
+          merge_from_csv(filename) do |row|
+            {
+              school_id:          row['school_id'],
+              school_year:        row['school_year'].to_i,
+              teaches_cs:         row['teaches_cs'],
+            }
+          end
+        end
+      end
+    end
+  end
+
+  # Loads/merges the data from a CSV into the table.
+  # Requires a block to parse the row.
+  # @param filename [String] The CSV file name.
+  # @param options [Hash] The CSV file parsing options.
+  # @param is_dry_run [Boolean] Allows testing of importing a CSV by rolling back any changes.
+  def self.merge_from_csv(filename, options = CSV_IMPORT_OPTIONS, is_dry_run: false)
+    census_summaries = nil
+    entries_skipped = 0
+
+    ActiveRecord::Base.transaction do
+      census_summaries = CSV.read(filename, **options).each do |row|
+        parsed = block_given? ? yield(row) : row.to_hash.symbolize_keys
```

Comment:
same thought here -- do you have a scenario in mind where you would pass a block? if not, this can just be:
```suggestion
        parsed = row.to_hash.symbolize_keys
```